api/server/sdk/credentials.go

```diff
@@ -589,3 +589,186 @@ func validateAndDeleteIfInvalid(ctx context.Context, s *CredentialServer, uuid s
 
 	return nil
 }
+
+// Update method modifies previously created credentials
+func (s *CredentialServer) Update(
+	ctx context.Context,
+	req *api.SdkCredentialUpdateRequest,
+) (*api.SdkCredentialUpdateResponse, error) {
+	if s.driver(ctx) == nil {
+		return nil, status.Error(codes.Unavailable, "Resource has not been initialized")
+	}
+
+	if len(req.GetCredentialId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply credentialId")
+	} else if aws := req.UpdateReq.GetAwsCredential(); aws != nil {
+		return s.awsUpdate(ctx, req.UpdateReq, aws, req.GetCredentialId())
+	} else if azure := req.UpdateReq.GetAzureCredential(); azure != nil {
+		return s.azureUpdate(ctx, req.UpdateReq, azure, req.GetCredentialId())
+	} else if google := req.UpdateReq.GetGoogleCredential(); google != nil {
+		return s.googleUpdate(ctx, req.UpdateReq, google, req.GetCredentialId())
+	}
+	return nil, status.Error(codes.InvalidArgument, "Unknown credential type")
+}
+
+func (s *CredentialServer) awsUpdate(
+	ctx context.Context,
+	req *api.SdkCredentialCreateRequest,
+	aws *api.SdkAwsCredentialRequest,
+	credId string,
+) (*api.SdkCredentialUpdateResponse, error) {
+
+	params := make(map[string]string)
+
+	params[api.OptCredType] = "s3"
```

Comment:
lets make provider type constant

Reply:
Since this is a cherry-pick, will leave this as is. lets address this in next PR in master.